src/rules/unit-no-unknown/index.js

```diff
@@ -0,0 +1,63 @@
+import { isString } from "lodash"
+import valueParser from "postcss-value-parser"
+
+import {
+  isKnownUnit,
+  cssWordIsVariable,
+  declarationValueIndexOffset,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "unit-no-unknown"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (u) => `Unexpected unknown unit "${u}"`,
+})
+
+export default function (actual, options) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual }, {
+      actual: options,
+      possible: {
+        ignore: [isString],
+      },
+      optional: true,
+    })
+
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      const { value } = decl
+
+      valueParser(value).walk(function (node) {
+        // Ignore wrong units within `url` function
+        if (node.type === "function" && node.value === "url") { return false }
+        if (node.type !== "word"
+          || cssWordIsVariable(node.value)
+        ) { return }
+
+        const parsedUnit = valueParser.unit(node.value)
+
+        if (!parsedUnit) { return }
+
+        const unit = parsedUnit.unit
+
+        if (!unit) { return }
+
+        const ignore = options && options.ignore || []
+
+        if (isKnownUnit(unit) || ignore.indexOf(unit) !== -1) { return }
```

Comment:
`optionHaveIgnored` is just a utility to do this. Lines 49-51 could be reduced to this:

```js
if (isKnownUnit(unit) || optionHaveIgnored(options, unit)) { return }
```